src/main/java/org/folio/oaipmh/helpers/AbstractGetRecordsHelper.java

```diff
@@ -193,78 +193,82 @@
-
-      Predicate<DataFieldType> generalInfoDataFieldPredicate = dataFieldType ->
-        dataFieldType.getInd1().equals(GENERAL_INFO_DATA_FIELD_INDEX_VALUE)
-        && dataFieldType.getInd2().equals(GENERAL_INFO_DATA_FIELD_INDEX_VALUE)
-        && dataFieldType.getTag().equals(GENERAL_INFO_DATA_FIELD_TAG_NUMBER);
-      records.forEach(recordType -> {
-        boolean isSuppressedFromDiscovery = recordType.isSuppressDiscovery();
-        String suppressDiscoveryValue = isSuppressedFromDiscovery ? "1" : "0";
-        gov.loc.marc21.slim.RecordType record = (gov.loc.marc21.slim.RecordType) recordType.getMetadata().getAny();
-        List<DataFieldType> datafields = record.getDatafields();
-        boolean alreadyContainsFolioSpecificDataField = datafields.stream()
-          .anyMatch(generalInfoDataFieldPredicate);
-
-        if(alreadyContainsFolioSpecificDataField) {
-          updateGeneralInfoDataFieldWithSuppressDiscoverySubfield(generalInfoDataFieldPredicate, suppressDiscoveryValue, datafields);
-        } else {
-          buildGeneralInfoDataFieldWithSuppressDiscoverySubfield(suppressDiscoveryValue, datafields);
-        }
-      });
+  private String updateSourceWithDiscoverySuppressedDataIfNecessary(String source, JsonObject sourceOwner, Request request) {
+    if(getBooleanProperty(request, REPOSITORY_SUPPRESSED_RECORDS_PROCESSING)) {
+      JsonObject content = new JsonObject(source);
+      JsonArray fields = content.getJsonArray(FIELDS);
+      Optional<JsonObject> generalInfoDataFieldOptional = getGeneralInfoDataField(fields);
+      if (generalInfoDataFieldOptional.isPresent()){
+        updateDatafieldWithDiscoverySuppressedData(generalInfoDataFieldOptional.get(), sourceOwner);
+      } else {
+        appendGeneralInfoDatafieldWithDiscoverySuppressedData(fields, sourceOwner);
+      }
+      return content.encode();
     }
+    return source;
   }
 
-  /**
-   * Updates instance general info data field (marked with "999" tag and both indexes have "f" value) with new subfield which
-   * holds data about record "suppress from discovery" state.
-   *
-   * @param generalInfoDataFieldPredicate - predicate with required field search criteria
-   * @param suppressDiscoveryValue - value that has to be assigned to subfield value field
-   * @param datafields - list of {@link DataFieldType} which contains folio specific data field
-   */
-  private void updateGeneralInfoDataFieldWithSuppressDiscoverySubfield(final Predicate<DataFieldType> generalInfoDataFieldPredicate, final String suppressDiscoveryValue, final List<DataFieldType> datafields) {
-    datafields.stream()
-      .filter(generalInfoDataFieldPredicate)
-      .findFirst()
-      .ifPresent(dataFieldType -> {
-        List<SubfieldatafieldType> subfields = dataFieldType.getSubfields();
-        subfields.add(new SubfieldatafieldType().withCode(INSTANCE_SUPPRESS_FROM_DISCOVERY_SUBFIELD_CODE)
-          .withValue(suppressDiscoveryValue));
-      });
+  @NotNull
```

Comment:
I'm not familiar with this, but a quick search suggests that this is specific to intellij IDEA?  Is that actually the case?  Does this have any impact at runtime?

Reply:
Hello, no it doesn't, i have deleted it